orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/graph/StageGraphBuilder.java

```diff
@@ -148,11 +148,18 @@ public void connect(@Nonnull Stage previous, @Nonnull Stage next) {
   }
 
   private String generateRefId() {
+    long offset = parent
+      .getExecution()
+      .getStages()
+      .stream()
+      .filter(i -> parent.getId().equals(i.getParentStageId()) && type == i.getSyntheticStageOwner())
```

Comment:
not necessary the most elegant, alternatives?

Reply:
Seems reasonable. I can't think of a neater way off the top of my head.